yb-voyager/src/lockfile/lockfile.go

```diff
@@ -0,0 +1,89 @@
+package lockfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/nightlyone/lockfile"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+type Lockfile struct {
+	fpath    string
+	cmdName  string
+	cmdPID   int
+	lockfile lockfile.Lockfile
+}
+
+func NewLockfile(fpath string) *Lockfile {
+	return &Lockfile{fpath: fpath, cmdPID: -1}
+}
+
+func (l *Lockfile) GetCmdName() string {
+	if l.cmdName != "" {
+		return l.cmdName
+	}
+
+	fname := filepath.Base(l.fpath)
+	l.cmdName = fname[1 : len(fname)-len("Lockfile.lck")]
+	l.cmdName = strings.Replace(l.cmdName, "-", " ", -1)
+	return l.cmdName
+}
+
+func (l *Lockfile) GetCmdPID() (int, error) {
+	if l.cmdPID != -1 {
+		return l.cmdPID, nil
+	}
+
+	bytes, err := os.ReadFile(l.fpath)
+	if err != nil {
+		return -1, fmt.Errorf("failed to read lockfile %q: %w", l.fpath, err)
+	}
+	l.cmdPID, err = strconv.Atoi(strings.Trim(string(bytes), " \n"))
+	if err != nil {
+		return -1, fmt.Errorf("failed to parse PID from lockfile %q: %w", l.fpath, err)
+	}
+	return l.cmdPID, nil
+}
+
+func (l *Lockfile) IsPIDActive() bool {
+	pid, err := l.GetCmdPID()
+	if err != nil {
+		return false
```

Comment:
How are we considering the error from `GetCmdPID` as `PID` not active?

Reply:
```
func (l *Lockfile) GetCmdPID() (int, error) {
	if l.cmdPID != -1 {
		return l.cmdPID, nil
	}

	bytes, err := os.ReadFile(l.fpath)
	if err != nil {
		return -1, fmt.Errorf("failed to read lockfile %q: %w", l.fpath, err)
	}
	l.cmdPID, err = strconv.Atoi(strings.Trim(string(bytes), " \n"))
	if err != nil {
		return -1, fmt.Errorf("failed to parse PID from lockfile %q: %w", l.fpath, err)
	}
	return l.cmdPID, nil
}
```

the case in which GetCmdPID() can fail is when it fails to read the pid from file which will happen when file doesn't exists because of either the command is not running or got exited/completed in the meantime.

Reply:
Got it! thanks.